src/mudlet-lua/lua/geyser/GeyserButton.lua

```diff
@@ -0,0 +1,244 @@
+-- @class table
+-- @name Geyser.Button
+-- @field twoState If true, the button will be treated as a two state button, with 'up' and 'down' states.
+-- @field state 'up' or 'down' depending on button state for two state buttons. Will always be 'up' if a single state button.
+-- @field tooltip The text to show when the button is hovered over. For two state buttons will be used for the 'up' tooltip
+-- @field downTooltip The text to show when the button is hovered over while in the 'down' state.
+-- @field toolTipDuration The amount of time for the tooltip to show. Used for both 'up' and 'down' states.
+-- @field clickCommand The command to send when the button is clicked. For two state buttons, used when clicking in the 'up' state. Will be skipped if clickFunction is defined.
+-- @field clickFunction Optional function to run when button clicked. For two state buttons, used when clicking in the 'up' state. If defined takes precedence over clickCommand.
+-- @field downCommand The command to send when the two state button is clicked while in the 'down' state.
+-- @field downFunction The function to run when the two state button is clicked while in the 'down' state.
+-- @field downColor The color to make the button while it is in the 'down' state. Superceded by downStyle
+-- @field downStyle The stylesheet to use for the button while it is in the 'down' state. Supercedes downColor
+-- @field color The color to make the button. If a two state button, will be used for the 'up' state. Superceded by style
+-- @field style The stylesheet to use for the button. If a two state button, will be used for the 'up' state. Supercedes color
+-- @field msg The text to put on the button. If a two state button, will be used for the 'up' state.
+-- @field downMsg The text to put on the button in the 'down' state.
+Geyser.Button = {
+  name            = "GeyserButtonClass",
+  height          = 50,
+  width           = 50,
+  color           = "blue",
+  downColor       = "blue",
+  msg             = "<center>Look</center>",
+  downMsg         = "<center>Look</center>",
+  tooltip         = "Click to look",
+  downTooltip     = "Click to look",
+  downCommand     = "look",
+  clickCommand    = "look",
+  twoState        = false,
+  state           = "up",
+  toolTipDuration = 5,
+}
+Geyser.Button.parent = Geyser.Label
+setmetatable(Geyser.Button, Geyser.Label)
+Geyser.Button.__index = Geyser.Button
+
+--- Creates a new button and returns it
+-- @param cons table of constraints which define the button
+-- @param container the container to place the button into
+function Geyser.Button:new(cons, container)
+  cons = cons or {}
+  local consType = type(cons)
+  if consType ~= "table" then
+    printError(f"bad argument #1 type (cons as table expects, got {consType})", true, true)
+  end
+  cons.name = cons.name or Geyser.nameGen("button")
+  local me = self.parent:new(cons, container)
+  setmetatable(me, self)
+  me:setClickCallback(function() me:press() end)
```

Comment:
Found one more place where the stacktrace is not super intuitive,

```
[ERROR:] object:<label callback event> function:<anonymous Lua function>
        <..._MudletuacscE/mudlet-lua/lua/geyser/GeyserButton.lua:52:  [string "Script: New 
script"]:21: attempt to concatenate global 'target' (a nil value)
stack traceback:
  ..._MudletuacscE/mudlet-lua/lua/geyser/GeyserButton.lua:118: in function 'press'
  ..._MudletuacscE/mudlet-lua/lua/geyser/GeyserButton.lua:52: in function <..._MudletuacscE/mudlet-
lua/lua/geyser/GeyserButton.lua:52>>
```

This happens with the default example when `target` is not defined

Reply:
In retrospect it does say `in function 'press'` somewhere in the middle, perhaps that'll be enough for folks to figure out what's happening.

Reply:
It does give the line that failed, though. ` [string "Script: New script"]:21: attempt to concatenate global 'target' (a nil value)`